File: conversation_memory.js
const fs = require('fs');
const path = 'brain/memory_log.json';

function loadMemory() {
    if (!fs.existsSync(path)) return [];
    return JSON.parse(fs.readFileSync(path, 'utf8'));
}

function saveConversation(user, jarvis) {
    const memory = loadMemory();
    memory.push({ timestamp: new Date().toISOString(), user, jarvis });
    fs.writeFileSync(path, JSON.stringify(memory, null, 2));
}

module.exports = { loadMemory, saveConversation };

File: conversation_memory.py
import json
from datetime import datetime
import os

LOG_FILE = "brain/memory_log.json"

def save_conversation(user, jarvis):
    timestamp = datetime.now().isoformat()
    entry = {
        "timestamp": timestamp,
        "user": user,
        "jarvis": jarvis
    }

    if not os.path.exists(LOG_FILE):
        with open(LOG_FILE, "w") as f:
            json.dump([entry], f, indent=4)
    else:
        with open(LOG_FILE, "r+") as f:
            data = json.load(f)
            data.append(entry)
            f.seek(0)
            json.dump(data, f, indent=4)

def load_memory():
    if not os.path.exists(LOG_FILE):
        return []

    with open(LOG_FILE, "r", encoding="utf-8") as f:
        return json.load(f)
